Automattic/newspack-newsletters
# feat: log send errors

### All Submissions:






### Changes proposed in this Pull Request:

Implements a new error-handling strategy for better debugging.

Newsletter sending errors should now be stored in a meta as an array, collecting up to 10 most recent errors. If any, these errors should be displayed in a sidebar panel:

<img width="286" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/dda0f8f1-6856-41e7-8cf6-5bd05481f1da">

The most recent error, if the newsletter is not sent, should also be displayed as an error notification in the editor:
 
<img width="754" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/afc0f289-c248-4b06-b549-7eee846e8125">

### How to test the changes in this Pull Request:

1. Check out this branch and configure your ESP to use Mailchimp
2. Add `return new WP_Error( 'some_error', 'Some error' );` to line `771` of `includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php`
3. Send a newsletter, and confirm you get the error immediate error: `Publishing failed. Some Error`
4. Refresh and confirm you get the persisted notification and the sidebar panel as shown above

### Other information:
